server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -319,6 +331,25 @@ private void operateCreateSnapshot() {
     IoTDB.metaManager.createMTreeSnapshot();
   }
 
+  private void operateKillQuery(KillQueryPlan killQueryPlan) throws QueryIdNotExsitException {
+    QueryTimeManager queryTimeManager = QueryTimeManager.getInstance();
+    if (killQueryPlan.getQueryId() != -1) {
+      if (queryTimeManager.getQueryThreadMap().get(killQueryPlan.getQueryId()) != null) {
+        queryTimeManager.killQuery(killQueryPlan.getQueryId());
+      } else {
+        throw new QueryIdNotExsitException(String
+            .format("Query Id %d is not exist, please check it.", killQueryPlan.getQueryId()));
+      }
+    } else {
+      // if queryId is not specified, kill all running queries
+      if (!queryTimeManager.getQueryThreadMap().isEmpty()) {
+        for (Long queryId : queryTimeManager.getQueryThreadMap().keySet()) {
+          queryTimeManager.killQuery(queryId);
+        }
```

Comment:
`killQuery` function will change the map while iterating the map keyset. This may cause `ConcurrentModificationException`.

Reply:
Fixed. Thanks for the reminder.